### PR TITLE
Update code to show filefeed sync events

### PR DESCRIPTION
**Pull request by mhpjay422**
Why this PR?
- On sync, show filefeed sync events on filefeed index

This PR contains: 
- Update to code to show filefeed sync events

### CODE REVIEW

**Comment by mhpjay422** [comment]
CleanShot 2023-06-23 at 11 25 02@2x
CleanShot 2023-06-23 at 12 19 54@2x

File: pages/api/flatfile/get-filefeed-actions.ts [hunk 1]
```diff
@@ -22,9 +22,19 @@ export default async function handler(
...
-  const actions = await getActions(
+  const eventActions = await getActions(
     token.organizationId,
     ActionType.FileFeedEvent
   );
+
+  const syncActions = await getActions(
+    token.organizationId,
+    ActionType.SyncFilefeedRecords
+  );
```
Comment (johnmosesman) [inline pages/api/flatfile/get-filefeed-actions.ts:33]: I would change getActions to take in a type [] and look up all of them in one query

File: lib/flatfile.ts [hunk 2]
```diff
@@ -62,8 +62,11 @@ export const createSpace = async ({
       metadata: {
         userId,
       },
+      guestAuthentication: ["shared_link"],
```
Comment (johnmosesman) [inline lib/flatfile.ts:65]: let's use the listener to define this

File: pages/file-feed.tsx [hunk 3]
```diff
@@ -75,18 +75,29 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
...
+  });
+
   const events = actions.map((a) => {
     const metadata = a.metadata as {
       topic: string;
     };
 
     return {
-      topic: metadata.topic,
+      topic: metadata.topic || null,
```
Comment (johnmosesman) [inline pages/file-feed.tsx:100]: should always have a topic here I think?

File: pages/file-feed.tsx [hunk 4]
```diff
@@ -75,18 +75,29 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
...
 
+  const syncActions = await getActions(
+    token.organizationId,
+    ActionType.SyncFilefeedRecords
+  );
+
+  const actions = [...eventActions, ...syncActions].sort((a, b) => {
+    return a.createdAt > b.createdAt ? -1 : 1;
+  });
+
```
Comment (johnmosesman) [inline pages/file-feed.tsx:93]: same here as above

File: lib/flatfile.ts (hunk 2 above)
Comment (mhpjay422) [reply lib/flatfile.ts:65]: removed

File: pages/file-feed.tsx (hunk 3 above)
Comment (mhpjay422) [reply pages/file-feed.tsx:100]: removed

File: pages/api/flatfile/get-filefeed-actions.ts (hunk 1 above)
Comment (mhpjay422) [reply pages/api/flatfile/get-filefeed-actions.ts:33]: Edited

File: pages/file-feed.tsx (hunk 4 above)
Comment (mhpjay422) [reply pages/file-feed.tsx:93]: Edited

File: lib/sync-records.ts [hunk 5]
```diff
@@ -172,9 +174,9 @@ export const syncWorkbookRecords = async ({
   await createAction({
     userId,
     organizationId,
-    type: ActionType.SyncRecords,
+    type: actionType,
     description: message,
-    metadata: {},
+    metadata: { topic: `Sync:${workflow} Records` },
```
Comment (johnmosesman) [inline lib/sync-records.ts:179]: Are the events from FF lowercase?

**Review by johnmosesman — APPROVED** [review]

**Pull request merged**